stack-graphs/src/c.rs

```diff
@@ -1576,17 +1576,45 @@ pub extern "C" fn sg_partial_path_arena_find_partial_paths_in_file(
     let partial_path_list = unsafe { &mut *partial_path_list };
     let cancellation_flag: Option<&AtomicUsize> =
         unsafe { std::mem::transmute(cancellation_flag.as_ref()) };
-    partials
-        .find_all_partial_paths_in_file(
-            graph,
-            file,
-            &AtomicUsizeCancellationFlag(cancellation_flag),
-            |_graph, partials, mut path| {
-                path.ensure_both_directions(partials);
-                partial_path_list.partial_paths.push(path);
-            },
-        )
-        .into()
+    sg_partial_path_arena_find_partial_paths_in_file_inner(
+        graph,
+        partials,
+        file,
+        partial_path_list,
+        &AtomicUsizeCancellationFlag(cancellation_flag),
+    )
+    .into()
+}
+
+fn sg_partial_path_arena_find_partial_paths_in_file_inner(
+    graph: &StackGraph,
+    partials: &mut PartialPaths,
+    file: Handle<File>,
+    partial_path_list: &mut sg_partial_path_list,
+    cancellation_flag: &dyn CancellationFlag,
+) -> Result<(), CancellationError> {
+    let mut db = Database::new();
```

Comment:
A core design pattern of the C API is that it is always the caller that is responsible for allocation our arena types.  This suggests that the publicly exported function should take in an `sg_partial_path_database` pointer.  (And for consistency with the rest of the API, should therefore be renamed from `sg_partial_path_arena_find_partial_paths_in_file` to `sg_partial_path_database_find_partial_paths_in_file`.)

Reply:
Also disregard this, since #184 removes the temporary DB that's only used for the duration of this call

Reply:
This will indeed be gone soon, but thanks for adding this context! That is helpful to know come future work that touches the C API.